### PR TITLE
Cfspdk 1115 add gecko adc driver

### PR DESCRIPTION
**Pull request by kimmansfield**

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -270,6 +270,11 @@ manifest:
       path: modules/lib/zcbor
     - name: zscilib
       path: modules/lib/zscilib
+      revision: ca070ddabdaf67175a2da901d0bd62e8899371c5
+    - name: dfu_gecko
+      url: git@github.com:tmobile/DevEdge-IoTDevKit-Zephyr-dfu-gecko.git
+      revision: ff3074ade2db045c95e72eb07ab089f70536274f
+      path: modules/hal/silabs/gecko/dfu_gecko
```
Comment (jtbaumann) [inline west.yml:277]: Souldn't be modifying west.yml in this commit

File: drivers/CMakeLists.txt [hunk 2]
```diff
@@ -72,3 +72,5 @@ add_subdirectory_ifdef(CONFIG_VIRTUALIZATION virtualization)
 add_subdirectory_ifdef(CONFIG_W1 w1)
 add_subdirectory_ifdef(CONFIG_WATCHDOG watchdog)
 add_subdirectory_ifdef(CONFIG_WIFI wifi)
+add_subdirectory_ifdef(CONFIG_SBS_BATTERY  fuel_gauge)
```
Comment (jtbaumann) [inline drivers/CMakeLists.txt:75]: This shouldn't be needed

File: boards/arm/tmo_dev_edge/tmo_dev_edge.dts [hunk 3]
```diff
@@ -102,6 +104,18 @@
 		regulator-name = "bq24250rge";
 		regulator-boot-on;
 	};
+
+	adc0:adc0@40002000 {
+		compatible = "sbs,sbs-battery";
```
Comment (jtbaumann) [inline boards/arm/tmo_dev_edge/tmo_dev_edge.dts:109]: Our adc shouldn't be named "sbs,sbs-battery"

File: dts/bindings/fuel-gauge/sbs,sbs-battery.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+
```
Comment (jtbaumann) [inline dts/bindings/fuel-gauge/sbs,sbs-battery.yaml:1]: This binding needs to be moved to adc

File: samples/drivers/adc/prj.conf [hunk 5]
```diff
@@ -1 +1,2 @@
 CONFIG_ADC=y
+CONFIG_ADC_SHELL=y
```
Comment (jtbaumann) [inline samples/drivers/adc/prj.conf:2]: Shouldn't be adding this to this sample

**Comment by ryanhagen-tmo** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| zscilib | https://github.com/zephyrproject-rtos/zscilib/commit/0035be5e6a45e4ab89755b176d305d7a877fc79c (master) | https://github.com/zephyrproject-rtos/zscilib/commit/ca070ddabdaf67175a2da901d0bd62e8899371c5 | [zephyrproject-rtos/zscilib@0035be5e..ca070dda](https://github.com/zephyrproject-rtos/zscilib/compare/0035be5e6a45e4ab89755b176d305d7a877fc79c..ca070ddabdaf67175a2da901d0bd62e8899371c5) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

File: drivers/CMakeLists.txt (hunk 2 above)
Comment (kimmansfield) [reply drivers/CMakeLists.txt]: Removed

File: samples/drivers/gecko_adc/src/main.c [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+	k_msleep(5000);
+
+	struct adc_sequence seq = {
+		.buffer = &buffer,
+		.buffer_size = sizeof(buffer),
+		.resolution = 12
+	};
+
+
```
Comment (johnlange2) [inline samples/drivers/gecko_adc/src/main.c:89]: extra blank line here

File: samples/drivers/gecko_adc/src/main.c [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+	uint8_t charging;
+	uint8_t vbus;
+	uint8_t battery_attached;
+	uint8_t fault;
+	uint8_t percent;
+	uint32_t millivolts = 0;
+
+	struct adc_gecko_data *data = batt_dev->data;
+
+
```
Comment (johnlange2) [inline samples/drivers/gecko_adc/src/main.c:61]: extra blank line here

File: boards/arm/tmo_dev_edge/tmo_dev_edge.dts [hunk 8]
```diff
@@ -11,6 +11,8 @@
 #include <dt-bindings/pwm/pwm.h>
 #include "tmo_dev_edge-pinctrl.dtsi"
 
+
```
Comment (johnlange2) [inline boards/arm/tmo_dev_edge/tmo_dev_edge.dts:14]: extra blank line

**Pull request closed without merging**